pkg/util/runtimeconfig/manager.go

```diff
@@ -55,14 +64,21 @@ type Manager struct {
 
 	configLoadSuccess prometheus.Gauge
 	configHash        *prometheus.GaugeVec
+
+	bucketClient        objstore.Bucket
+	bucketClientFactory BucketClientFactory
 }
 
 // New creates an instance of Manager and starts reload config loop based on config
-func New(cfg Config, registerer prometheus.Registerer, logger log.Logger) (*Manager, error) {
+func New(cfg Config, registerer prometheus.Registerer, logger log.Logger, factory BucketClientFactory) (*Manager, error) {
 	if cfg.LoadPath == "" {
 		return nil, errors.New("LoadPath is empty")
 	}
 
+	if cfg.StorageConfig.Backend == "" {
+		cfg.StorageConfig.Backend = bucket.Filesystem
```

Comment:
Shall we add a log about the default to filesystem? I am okay to even throw an error.

Reply:
A log seems fine to me. In which case do you think we should throw an error? For backwards compatibility I was thinking of defaulting to filesystem since that is what Cortex does currently.

Reply:
Just curious if we should do the default or throw an error instead.
We set the default to `filesystem` already but if backend is still empty, that means user changes it to empty string and I think it is not valid. We should probably throwing an error here.

And I guess even if we don't throw the error, it will probably fail when we initialize the bucket client. 
Question here is, do we need to do the default to filesystem?

Reply:
Ah I see what you mean, because we set the default backend to filesystem when we register the flags we don't want users to make an explicit invalid configuration. That makes sense, I'll update this to return an error instead.